frame/offences/benchmarking/src/lib.rs

```diff
@@ -151,25 +178,162 @@ benchmarks! {
+		// make sure that all slashes have been applied
+		assert_eq!(
+			System::<T>::event_count(), 0
+			+ 1 // offence
+			+ 2 * r // reporter (reward + endowment)
+			+ o // offenders slashed
+			+ o * n // nominators slashed
+		);
+	}
+
+	report_offence_grandpa {
+		let r in 1 .. MAX_REPORTERS;
+		let n in 0 .. MAX_NOMINATORS.min(MAX_NOMINATIONS as u32);
+		let o = 1;
+
+		// Make r reporters
+		let mut reporters = vec![];
+		for i in 0 .. r {
+			let reporter = account("reporter", i, SEED);
+			reporters.push(reporter);
+		}
+
+		// make sure reporters actually get rewarded
+		Staking::<T>::set_slash_reward_fraction(Perbill::one());
+
+		let mut offenders = make_offenders::<T>(o, n).expect("failed to create offenders");
+		let keys =  ImOnline::<T>::keys();
+
+		let offence = GrandpaEquivocationOffence {
+			time_slot: GrandpaTimeSlot { set_id: 0, round: 0 },
+			session_index: 0,
+			validator_set_count: keys.len() as u32,
+			offender: T::convert(offenders.pop().unwrap()),
+		};
+		assert_eq!(System::<T>::event_count(), 0);
+	}: {
+		let _ = Offences::<T>::report_offence(reporters, offence);
+	}
+	verify {
+		// make sure the report was not deferred
+		assert!(Offences::<T>::deferred_offences().is_empty());
+		// make sure that all slashes have been applied
+		assert_eq!(
+			System::<T>::event_count(), 0
+			+ 1 // offence
+			+ 2 * r // reporter (reward + endowment)
+			+ o // offenders slashed
+			+ o * n // nominators slashed
+		);
+	}
+
+	report_offence_babe {
+		let r in 1 .. MAX_REPORTERS;
+		let n in 0 .. MAX_NOMINATORS.min(MAX_NOMINATIONS as u32);
+		let o = 1;
+
+		// Make r reporters
+		let mut reporters = vec![];
+		for i in 0 .. r {
+			let reporter = account("reporter", i, SEED);
+			reporters.push(reporter);
+		}
+
+		// make sure reporters actually get rewarded
+		Staking::<T>::set_slash_reward_fraction(Perbill::one());
+
+		let mut offenders = make_offenders::<T>(o, n).expect("failed to create offenders");
+		let keys =  ImOnline::<T>::keys();
+
+		let offence = BabeEquivocationOffence {
+			slot: 0,
+			session_index: 0,
+			validator_set_count: keys.len() as u32,
+			offender: T::convert(offenders.pop().unwrap()),
+		};
+		assert_eq!(System::<T>::event_count(), 0);
+	}: {
+		let _ = Offences::<T>::report_offence(reporters, offence);
+	}
+	verify {
+		// make sure the report was not deferred
+		assert!(Offences::<T>::deferred_offences().is_empty());
+		// make sure that all slashes have been applied
+		assert_eq!(
+			System::<T>::event_count(), 0
+			+ 1 // offence
+			+ 2 * r // reporter (reward + endowment)
+			+ o // offenders slashed
+			+ o * n // nominators slashed
+		);
+	}
 
 	on_initialize {
-		let d in ...;
+		let d in 1 .. MAX_DEFERRED_OFFENCES;
+		let o = 10;
+		let n = 100;
 
 		Staking::<T>::put_election_status(ElectionStatus::Closed);
 
 		let mut deferred_offences = vec![];
+		let offenders = make_offenders::<T>(o, n).expect("failed to create offenders");
```

Comment:
```suggestion
		let offenders = make_offenders::<T>(o, n)?;
```

I have this pattern way more. though If we do one iteration in each runtime then it makes no difference. A panic would just go to next iteration as well.